client/checkout/blocks/saved-token-handler.js

```diff
@@ -1,15 +1,41 @@
 /**
  * Internal dependencies
  */
+import { useEffect } from 'react';
 import { usePaymentCompleteHandler } from './hooks';
+import { useSelect } from '@wordpress/data';
 
 export const SavedTokenHandler = ( {
 	api,
 	stripe,
 	elements,
-	eventRegistration: { onCheckoutSuccess },
+	eventRegistration: { onPaymentSetup, onCheckoutSuccess },
 	emitResponse,
 } ) => {
+	const paymentMethodData = useSelect( ( select ) => {
+		const store = select( 'wc/store/payment' );
+		return store.getPaymentMethodData();
+	} );
+
+	useEffect( () => {
+		onPaymentSetup( async () => {
```

Comment:
Non-blocking: we can remove `await` from this function.

Reply:
Addressed it in [6dead9c](https://github.com/Automattic/woocommerce-payments/pull/7701/commits/6dead9cb960647545e583cd8e472f8e8882bf02a), thanks for the review!